packages/fixie/src/voice.ts

```diff
@@ -110,13 +118,20 @@ export class VoiceSession {
   /** Warm up the VoiceSession by making network connections. This is called automatically when the VoiceSession object is created. */
   warmup(): void {
     console.log('[voiceSession] warming up');
+    this.changeState(VoiceSessionState.CONNECTING);
```

Comment:
I'd trigger this after we create the socket

Reply:
So, that leads to a situation where if you get an error on socket connect, the state of the VoiceSession will not always be in the same state. It might be DISCONNECTED (if the error happens soon after socket creation) or CONNECTING (if the error happens a bit later). I thought it would be better to be consistent so one dealing with errors doesn't need to understand why it might be one state or another.